tests/sentry/tasks/test_base.py

```diff
@@ -233,3 +272,55 @@ def task_that_raises_retry_error():
     mock_task_state.retries_remaining = True
     with pytest.raises(RetryError):
         task_that_raises_retry_error()
+
+
+def test_instrumented_task_with_alias_same_namespace() -> None:
+    assert test_tasks.contains("tests.tasks.test_base.primary_task")
+    task_result = task_with_alias("test")
+    assert task_result == "Task with alias test"
+
+    assert test_tasks.contains("tests.tasks.test_base.alias_task")
+    alias_task_result = test_tasks.get("tests.tasks.test_base.alias_task")("test")
+    assert alias_task_result == "Task with alias test"
+
+    assert task_result == alias_task_result
+
+
+def test_instrumented_task_with_alias_different_namespaces() -> None:
+    assert test_tasks.contains("tests.tasks.test_base.primary_task_primary_namespace")
+    task_result = task_with_alias_and_alias_namespace("test")
+    assert task_result == "Task with alias and alias namespace test"
+
+    assert exampletasks.contains("tests.tasks.test_base.alias_task_alias_namespace")
+    alias_task_result = exampletasks.get("tests.tasks.test_base.alias_task_alias_namespace")("test")
+    assert alias_task_result == "Task with alias and alias namespace test"
+
+    assert task_result == alias_task_result
+
+
+@override_settings(SILO_MODE=SiloMode.REGION)
+def test_instrumented_task_with_alias_silo_limit_call_region() -> None:
+    assert test_tasks.contains("tests.tasks.test_base.region_primary_task")
+    assert region_task_with_alias("test") == "Region task with alias test"
+
+    assert test_tasks.contains("tests.tasks.test_base.region_alias_task")
+    assert region_task_with_alias("test") == "Region task with alias test"
```

Comment:
```suggestion
    assert test_tasks.get("tests.tasks.test_base.region_alias_task")("test") == "Region task with alias test"
```

The bot is right, fetching from the namespace would be better.